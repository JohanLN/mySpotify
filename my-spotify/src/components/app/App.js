import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { increment, decrement } from '../../redux/actions/counterActions';

function App() {

  const counter = useSelector(state => state.counter);
  const dispatch = useDispatch();

  return (
    <div className="App">
      <h1>Hello !</h1>

      <h2>Counter = {counter}</h2>

      <button onClick={() => dispatch(increment(1))}>+</button>
      <button onClick={() => dispatch(decrement(2))}>-</button>

    </div>
  );
}

export default App;
