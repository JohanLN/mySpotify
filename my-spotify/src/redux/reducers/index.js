import counterReducers from './counter';
import { combineReducers } from 'redux';

const rootReducers = combineReducers({
    counterReducers: counterReducers
});

export default rootReducers;