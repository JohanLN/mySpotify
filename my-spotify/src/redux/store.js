import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducers from './reducers/index';

export const storage = createStore(
    rootReducers,
    applyMiddleware(thunk),  
);