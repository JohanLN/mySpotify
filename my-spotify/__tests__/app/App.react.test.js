import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../src/components/app/App';
import Adapter from "enzyme-adapter-react-16"
import { shallow, configure } from "enzyme";
import { createStore } from 'redux';
import allReducers from '../../src/redux/reducers/index';
import { Provider } from 'react-redux';

const storage = createStore(
  allReducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

configure({adapter: new Adapter()});

describe("Tests of App.js", () => {

    it("do a simple snaptshot of App.js", () => {
        const provider = renderer.create(<Provider store={storage}><App/></Provider>)
    
        let snapShot = provider.toJSON();
        expect(snapShot).toMatchSnapshot();
    })

})
