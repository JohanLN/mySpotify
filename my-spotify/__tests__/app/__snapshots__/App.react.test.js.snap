// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple test of App.js 1`] = `
<div
  className="App"
>
  <h1>
    Hello !
  </h1>
  <h2>
    Counter = 
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Tests of App.js do a simple snaptshot of App.js 1`] = `
<div
  className="App"
>
  <h1>
    Hello !
  </h1>
  <h2>
    Counter = 
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
